vm/src/stdlib/itertools.rs

```diff
@@ -0,0 +1,212 @@
+        let result = self.cur.borrow().clone();
+        AddAssign::add_assign(&mut self.cur.borrow_mut() as &mut BigInt, &self.step);
+        Ok(PyInt::new(result))
+    }
+
+    #[pymethod(name = "__iter__")]
+    fn iter(zelf: PyRef<Self>, _vm: &VirtualMachine) -> PyRef<Self> {
+        zelf
+    }
+}
+
+#[pyclass]
+#[derive(Debug)]
+struct PyItertoolsRepeat {
+    object: PyObjectRef,
+    times: Option<RefCell<BigInt>>,
+}
+
+impl PyValue for PyItertoolsRepeat {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("itertools", "repeat")
+    }
+}
+
+#[pyimpl]
+impl PyItertoolsRepeat {
+    #[pymethod(name = "__new__")]
+    fn new(
+        _cls: PyClassRef,
+        object: PyObjectRef,
+        times: OptionalArg<PyIntRef>,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        let times = match times.into_option() {
+            Some(int) => Some(RefCell::new(int.as_bigint().clone())),
+            None => None,
+        };
+
+        Ok(PyItertoolsRepeat {
+            object: object.clone(),
+            times: times,
+        }
+        .into_ref(vm)
+        .into_object())
+    }
+
+    #[pymethod(name = "__next__")]
+    fn next(&self, vm: &VirtualMachine) -> PyResult {
+        if self.times.is_some() {
+            match self.times.as_ref().unwrap().borrow().cmp(&BigInt::from(0)) {
+                Ordering::Less | Ordering::Equal => return Err(new_stop_iteration(vm)),
+                _ => (),
+            };
+
+            SubAssign::sub_assign(
+                &mut self.times.as_ref().unwrap().borrow_mut() as &mut BigInt,
+                &BigInt::from(1),
+            );
+        }
+
+        Ok(self.object.clone())
+    }
+
+    #[pymethod(name = "__iter__")]
+    fn iter(zelf: PyRef<Self>, _vm: &VirtualMachine) -> PyRef<Self> {
+        zelf
+    }
+}
+
+#[pyclass]
+#[derive(Debug)]
+struct PyItertoolsTakewhile {
+    predicate: PyObjectRef,
+    iterable: PyObjectRef,
+    stop_flag: RefCell<bool>,
+}
+
+impl PyValue for PyItertoolsTakewhile {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("itertools", "takewhile")
+    }
+}
+
+#[pyimpl]
+impl PyItertoolsTakewhile {
+    #[pymethod(name = "__new__")]
+    fn new(
+        _cls: PyClassRef,
+        predicate: PyObjectRef,
+        iterable: PyObjectRef,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        let iter = match get_iter(vm, &iterable) {
+            Ok(iter) => iter,
+            Err(err) => return Err(err),
+        };
+
+        Ok(PyItertoolsTakewhile {
+            predicate: predicate,
+            iterable: iter,
+            stop_flag: RefCell::new(false),
+        }
+        .into_ref(vm)
+        .into_object())
+    }
+
+    #[pymethod(name = "__next__")]
+    fn next(&self, vm: &VirtualMachine) -> PyResult {
+        if *self.stop_flag.borrow() {
+            return Err(new_stop_iteration(vm));
+        }
+
+        let obj = call_next(vm, &self.iterable);
```

Comment:
Please use the `?` operator here, like this:
```rust
let obj = call_next(vm, &self.iterable)?;
```
The match statement can be replaced by the `?` suffix.